roxie/ccd/ccdserver.cpp

```diff
@@ -25207,7 +25207,7 @@ class CRoxieServerFullKeyedJoinHead: public CRoxieServerActivity, implements IRe
 
         // called from front puller thread
         // buffer up an IndexRead request
-        if (helper.leftCanMatch(row))
+        if (keySet && helper.leftCanMatch(row))
```

Comment:
I thought about it but it's such crazy usage I decided I didn't care (keyed join where the index exists but the rhs file does not, and was specified as OPT).